src/storage/v2/inmemory/storage.cpp

```diff
@@ -2687,6 +2683,52 @@ void InMemoryStorage::CreateSnapshotHandler(
     });
   }
 }
+
+std::optional<std::tuple<EdgeRef, EdgeTypeId, Vertex *, Vertex *>> InMemoryStorage::FindEdge(Gid gid) {
+  using EdgeInfo = std::optional<std::tuple<EdgeRef, EdgeTypeId, Vertex *, Vertex *>>;
+
+  auto edge_acc = edges_.access();
+  auto edge_it = edge_acc.find(gid);
+  if (edge_it == edge_acc.end()) {
+    return std::nullopt;
+  }
+
+  auto *edge_ptr = &(*edge_it);
```

Comment:
I assume `DMG_ASSERT(config_.salient.items.properties_on_edges == false);` at this point.

Reply:
You mean `== true`, right?